aten/src/ATen/NestedTensorImpl.h

```diff
@@ -167,9 +161,10 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   //    && nesting in ascending order
   const at::Tensor storage_offsets_;
   // NOTE: -1 here means the size is missing
+  // Optional to allow it to be computed lazily from nested
   // TODO: maybe we can remove this metadata since
   //       we can compute it from `nested_sizes_`
-  std::vector<int64_t> opt_sizes_;
+  c10::optional<std::vector<int64_t>> opt_sizes_;
```

Comment:
Hm what would be the benefits of that approach over this one?

Reply:
Instead of assigning a object when this is being populated, we'd assign values (entries). Also it'd enforce that the size of opt_sizes_ is derived from dim.